src/vcf/record.cpp

```diff
@@ -259,6 +273,17 @@ namespace ebi
        }
     }
     
+    void Record::check_info_no_duplicates() const
+    {
+        if (source->version == Version::v43 && info.size() > 1) {
+            for (auto & in : info) {
+                if (info.count(in.first) > 1) {
+                    throw new InfoBodyError{line, "INFO must not have duplicate keys"};
```

Comment:
Another way we could do it is using the `Record::check_no_duplicates()` method. But for that we'd have to use a vector for the keys and then construct the counting map. So I think using this is better instead.